pkg/render/logstorage.go

```diff
@@ -1419,3 +1395,60 @@ func (es elasticsearchComponent) kibanaPodSecurityPolicy() *policyv1beta1.PodSec
 	psp.GetObjectMeta().SetName("tigera-kibana")
 	return psp
 }
+
+// If user has set ResourceRequirements, replace individual field's default value with user's value.
+// If user provided both Limits and Requests, use them.
+// As ResourceRequirements.Limits must be >= ResourceRequirements.Requests,
+// if user provided just Limits, and Limits is <= default Requests, set Requests value as user's Limits value,
+// if user provided just Requests, and Requests is >= default Limits, set Limits value as user's Requests value.
+func overrideResourceRequirements(defaultReq *corev1.ResourceRequirements, userOverrides corev1.ResourceRequirements) {
+	if userOverrides.Limits.Cpu().Value() != 0 {
+		defaultReq.Limits["cpu"] = *userOverrides.Limits.Cpu()
+		if userOverrides.Requests.Cpu().Value() == 0 && defaultReq.Requests.Cpu().Value() > userOverrides.Limits.Cpu().Value() {
```

Comment:
I don't think we should overwrite the value if the user has specifically set the limit. If the user sets a bad value we should either set it as that's there intent, or fail so they know that something went wrong. 

If kuberentes does not allow a limit to be smaller the the requests than it will fail so we don't need to fail. If kuberenets does allow it then we should allow it as it may be the users intent.

Reply:
We chatted an I had a misunderstanding about this, the limits are only updated if there's requests but no limits set.